sqlmesh/core/engine_adapter/spark.py

```diff
@@ -428,6 +431,19 @@ def _create_table(
         column_descriptions: t.Optional[t.Dict[str, str]] = None,
         **kwargs: t.Any,
     ) -> None:
+        table_name = (
+            table_name_or_schema.this
+            if isinstance(table_name_or_schema, exp.Schema)
+            else exp.to_table(table_name_or_schema)
+        )
+        # Spark doesn't support creating a wap table DDL. Therefore we check if this is a wap table and if it is,
+        # this is not a replace, and the table already exists then we can safely just return. Otherwise we let it error.
+        if not expression and isinstance(table_name.this, exp.Dot):
+            wap_id = table_name.this.parts[-1].name
+            if wap_id.startswith(f"{self.BRANCH_PREFIX}{self.WAP_PREFIX}"):
+                table_name.set("this", table_name.this.this)
+
+        already_exists = False if not exists else self.table_exists(table_name)
```

Comment:
Does it mean we now check table existence on each table creation call? Should we change `exists` to False` by default then?

Reply:
Also why call the parent method if the table exists?

Reply:
Can we at least set it to False when creating tables in the snapshot evaluator since we already check table existence there.

Reply:
Also I feel like the check is only relevant if it's iceberg.

Reply:
Now does check only if WAP supported and `IF NOT EXISTS`: https://github.com/TobikoData/sqlmesh/pull/2033/commits/ae6bb0e329b0f48285ccb1277db3e3601c755cdb

Can you provide more context for this feedback? I don't follow.

>Also why call the parent method if the table exists?

>Can we at least set it to False when creating tables in the snapshot evaluator since we already check table existence there.